### ISSUE TITLE
support for flickr ?

### ISSUE THREAD

**Issue by emmtte** [rclone/rclone#52]
Is there plan for the `flickr` support in rclone ~~for the free 1TB space~~ ?

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by zeshanb** [comment]
Hello Emmanuel,

rClone is a software you can install on your system and run by passing
simple commands to backup your online storage data.

With that being said. If flickr provides an API allowing users to download
their images from flickr's object storage. It could be possible.

Regards,
Zeshan

**Comment by ncw** [comment]
I don't know what exactly you can store in flickr's free space, but it might be possible!

Any pointers to APIs?

**Comment by ManuCart** [comment]
To store pics and videos for backup utility
[https://www.flickr.com/services/api/]

**Comment by zeshanb** [comment]
Are you interested in Add photo post API for flickr?

https://www.flickr.com/services/api/flickr.photosets.addPhoto.html

Regards,
Zeshan

On Wed, Apr 29, 2015 at 2:03 PM, Emmanuel CHARETTE <notifications@github.com

**Comment by ncw** [comment]
From a quick look
  * could store md5sums in a tag
  * has ability to set dates
  * Might have to map directories to tags, or maybe photo sets

So probably could make an rclone interface...

**Comment by ManuCart** [comment]
Great news :), hope is useful for rclone others users, and 1TB for free could chance to try.

**Comment by clintbxtreme** [comment]
This would be awesome! Please implement.

**Comment by andrewbecks** [comment]
As an rclone user, I'm adding a thumbs up to this idea!

**Comment by markjamie** [comment]
Definitely, definitely in support of this idea! Flickr's 1TB is great for photo and video storage. Rclone support would allow for local copies on a NAS and cloud copies in Flickr

**Comment by githubhjs** [comment]
recently, flickr announced that it will no longer provide 1TB free space and gonna remove those photos beyond 1000 within 3 months. so this enhancement is gonna be more important.
thanks!

**Comment by snipem** [comment]
@githubhjs In contrast I would say Flickrs restriction makes the extension **less** important. For a one time download of all your files other measurements can be taken (https://help.flickr.com/en_us/download-photos-or-albums-in-flickr-HJeLjhQskX).

Since this restriction renders Flickr as a free picture host useless I recommend closing this issue.

**Comment by ManuCart** [comment]
So bad, can close this issue

**Comment by eugenevd** [comment]
I disagree with closing this issue.
There are people who have paid accounts, who would like to access the files there via something like rclone

**Comment by ameuret** [comment]
Closing this feature request because Flickr's free tier has become less generous does not make any sense. Most third-party services for which rclone has integrations have free and paid plans.
If I was a Go man I'd offer some coding time. Unfortunately, Go is not on my tool belt yet.

All I can for now do is offer to test dev branches on  my paid Flickr account.

**Comment by dredwolff** [comment]
Yes, please implement this! I don't know Go yet either, but I just downloaded rclone hoping to sync some data between google and flickr.  Do you need money? I can send money, or beer, or even liquor 😂

**Comment by ivandeex** [comment]
Gonna hide most comments as they can be squashed to simple +1 or -1 on the top post.
If you really want this backend and know Go, just **submit a pull request**. We'll review it.
If you want this and **don't** know Go, put a thumbs up at the top.
That's all.

**Comment by ivandeex** [comment]
If anybody here really wants to help this issue, they can at least test out flickr-related projects on github:
- https://github.com/masci/flickr/tree/v2 - abandoned for 5 years but go-based
- https://github.com/mncaudill/go-flickr - ditto
- https://github.com/mikf/gallery-dl - maintained but python based, cannot be used directly in rclone

**Comment by ivandeex** [comment]
If anybody here really wanted to help this issue, they can at least **test out flickr-related projects** on github (`go get github.com/denmark/flickr`,  then try to list/download a photoset) and report below:
- https://github.com/masci/flickr/tree/v2 - abandoned for 5 years but go-based :-1:
- https://github.com/denmark/flickr - continuation of the one above, last commit only 5 months ago :+1:, needs testing, go based
- https://github.com/mikf/gallery-dl - maintained but python based, cannot be used directly in rclone, so unlikely to be used but can borrow ideas from :thinking:
- https://github.com/mncaudill/go-flickr - abandoned for 5 years :-1:

Volunteers, where are you?

P.S. Free plan limits: [<1000 photos/videos](https://www.flickrhelp.com/hc/en-us/articles/4404070279700-Get-Started-with-Flickr) & [size/format limits per photo/video](https://www.flickrhelp.com/hc/en-us/articles/4404079649300-Flickr-upload-requirements).

**Comment by ivandeex** [comment]
The above note targeted testers.

This one is dedicated for backend developers.
 A comment https://github.com/rclone/rclone/issues/2267#issuecomment-383513676 by @ncw fully pertains herein:

> The unit tests would need a lot of work if it is only possible to upload images and videos - I think that would be a bit of a drawback as I normally lean very heavily on the unit tests when bringing up a new backend.
>
> For a first step, I'd find out how many of the [features](https://rclone.org/overview/#features) and [optional features](https://rclone.org/overview/#optional-features) it could support.  Of particular interest - can it set a last modified time and does it support a content hash...

**Comment by ivan98** [comment]
Now that Google has announced it is closing G Suite Legacy free edition, folks will be looking to move, and Flickr Pro is an option.

**Comment by BigBlueHat** [comment]
I mainly want to bring my Flickr Pro plan photos locally (for additional backup). So the price of the plan is immaterial (though I'm keen to stop having to pay it...and get my data back...).

> Volunteers, where are you?
I'd love to say I'm volunteering...but my time hasn't (yet) been able to go this way... Maybe sometime this winter.

I'd certainly be happy to test if someone else puts something together.

**Comment by ncw** [comment]
I'm going to close this now as we have had no volunteers in the last 10 years! If someone wants to make a flickr backend we can re-open.
